ee/codegen/src/generators/workflow.ts

```diff
@@ -416,19 +444,59 @@ export class Workflow {
         initializer: python.TypeInstantiation.dict(
           this.workflowContext.workflowOutputContexts.map(
             (workflowOutputContext) => {
-              const terminalNodeData =
+              const finalOutput =
                 workflowOutputContext.getFinalOutputNodeData();
+              let outputNodeId: string;
+              let targetHandleId: string;
+              let outputId: string;
+              let nodeInputId: string | undefined;
+              let name: string;
+              let label: string;
+              let displayData: NodeDisplayDataType | undefined;
+
+              // Final output node
+              if ("type" in finalOutput) {
+                outputNodeId = finalOutput.id;
+                targetHandleId = finalOutput.data.targetHandleId;
+                outputId = finalOutput.data.outputId;
+                nodeInputId = finalOutput.data.nodeInputId;
+                name = finalOutput.data.name;
+                label = finalOutput.data.label;
+                displayData = finalOutput.displayData;
+              } else {
+                // Workflow output value
+                const referencedNode = this.workflowContext.getNodeContext(
+                  finalOutput.value.nodeId
+                );
+                if (!referencedNode) {
+                  throw new WorkflowGenerationError(
+                    `Could not find node ${finalOutput.value.nodeId}`
+                  );
+                }
+                const referencedOutput =
+                  this.workflowContext.getOutputVariableContextById(
+                    finalOutput.outputVariableId
+                  );
+                outputNodeId = referencedNode.getNodeId();
+                targetHandleId = this.getTargetHandleId(
+                  referencedNode.nodeData
+                );
+                outputId = finalOutput.value.nodeOutputId;
+                name = referencedOutput.name;
+                label = this.getLabel(referencedNode.nodeData);
+                displayData = referencedNode.nodeData.displayData;
+              }
 
               const edge = this.getEdges().find((edge) => {
```

Comment:
[oos] We may also want to split out a refactor that removes `edge_id` from `WorkflowOutputVellumDisplayOverrides`. It's only used to eventually fill in this value: https://github.com/vellum-ai/vellum-python-sdks/blob/main/ee/vellum_ee/workflows/display/workflows/vellum_workflow_display.py#L185. But, by nature of the edge being synthetic, we are likely fine to depend only on synthetic ids for a synthetically made edge